training/save_load_model_to_file.py

```diff
@@ -0,0 +1,30 @@
+import joblib
+
+from training.create_model import get_fitted_model
+from training.creating_dataset import create_dataframe
+
+
+def save_model_to_file(model=None, filename: str = "../saved_models/current_model.pkl"):
+    """
+    Saves a machine learning model to a file
+    :param model: machine learning model to be saved. If None, default model will be created and trained
```

Comment:
To jest baaaaardzo zły pomysł. Mamy zasady SOLID, z czego moim zdaniem absolutnie najważniejsza jest już pierwsza: Single Responsibility Principle (SRP). Funkcja czy metoda powinna robić jedną rzecz. Implicite zakłada się też, że jest to oczywiste na podstawie jej nazwy.

Mamy funkcję `save_model_to_file` - logiczne, zapisuje podany model do pliku pod podaną ścieżką. Typowa funkcja, która ma side effecty, których nie lubimy (bo programowanie funkcyjne ma parę dobrych pomysłów, i to jest jeden z nich), dlatego zamykamy I/O w osobnej funkcji.

Co robi ta funkcja, jeżeli nie zostanie podany model?
1. Wczytuje zbiór danych, sięga do innych plików i je wczytuje przy okazji
2. Preprocessuje zbiór danych
3. Trenuje model
4. Zapisuje go

To jest połowa (albo i więcej) całego waszego kodu wywoływana w momencie, jeżeli ktoś poda `model=None`. Pamiętajcie, że to użytkownik kodu jest odpowiedzialny za podanie sensownych wartości. Logicznie nie ma sensu sytuacja, że ktoś wywołuje funkcję zapisującą model, i nie podaje modelu do zapisania.